### PR TITLE
USHIFT-1588: Remove rebase-payload

### PR DESCRIPTION
**Pull request by pacevedom**
Remove the rebase-payload image by handling dependencies in the CI container where it runs, instead of preparing them upfront.
With this every test job will save around 5min on average, since there is no need to build an image that its not going to be used. Rebase job is also optimized as a byproduct.

### CODE REVIEW

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-main-test-rebase pull-ci-openshift-microshift-release-4.14-test-rebase

File: ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ref:
   as: openshift-microshift-rebase
-  from: rebase-payload
+  from: src
```
Comment (pmtk) [inline ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml:3]: Is `rhel-9-release-golang-1.20-openshift-4.14` base for `src`?
I'm thinking ahead when we bump that and perhaps some day common code in `commands.sh` will need some tweaking (or some python libraries get updated so our code will work with older but not with newer), but we can deal with that in (far?) future

**Comment by pmtk** [comment]
/lgtm
/hold

Unhold when convenient

File: ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml (hunk 1 above)
Comment (pacevedom) [reply ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml:3]: Yes, `src` image comes from `root` image, which in turn comes from `build_root`, which is updated everytime we get a new go version (or ocp version, for that matter).

**Pull request merged**